aspen/website.py

```diff
@@ -68,15 +69,19 @@ def redirect(self, location, code=None, permanent=False, base_url=None, response
         permanent is True and 302 (Found) if it is False. If url doesn't start
         with base_url (defaulting to self.base_url), then we prefix it with
         base_url before redirecting. This is a protection against open
-        redirects. If you provide your own response we will set .code and
-        .headers['Location'] on it.
+        redirects. If base_url is not '' and location is not an absolute path,
```

Comment:
Well, but I think you've still got the logic backwards: `base_url` cannot *not* be empty if you wish to use relative paths. If you pass a relative path such as `../foo` for location and a non-empty `base_url` such as `http://example.com`, then you're asking for a computed location of `http://example.com../foo`, which is garbage. No?

Instead, you should override any globally configured `base_url` by passing `base_url=''` in a specific `redirect` call.

```python
website.base_url = 'http://example.com'
website.redirect('../foo', base_url='')
```